src/main/java/games/strategy/triplea/ui/TripleAFrame.java

```diff
@@ -264,15 +264,15 @@ public void windowClosing(final WindowEvent e) {
     chatSplit.setOneTouchExpandable(true);
     chatSplit.setDividerSize(8);
     chatSplit.setResizeWeight(0.95);
-    if (GameRunner.hasChat()) {
+    if (GameRunner.getChat().isPresent()) {
```

Comment:
NB: For future reference, we have an `OptionalUtils#ifPresentOrElse()` "polyfill" for the `Optional#ifPresentOrElse()` method added in Java 9.